modules/imgproc/src/subdivision2d.cpp

```diff
@@ -210,19 +210,130 @@ void Subdiv2D::swapEdges( int edge )
     splice(sedge, getEdge(b, NEXT_AROUND_LEFT));
 }
 
-static double triangleArea( Point2f a, Point2f b, Point2f c )
-{
-    return ((double)b.x - a.x) * ((double)c.y - a.y) - ((double)b.y - a.y) * ((double)c.x - a.x);
+static double CrossProduct(Point2f u, Point2f v) {
+    return (double)u.x * v.y - (double)v.x * u.y;
 }
 
-int Subdiv2D::isRightOf(Point2f pt, int edge) const
-{
-    Point2f org, dst;
-    edgeOrg(edge, &org);
-    edgeDst(edge, &dst);
-    double cw_area = triangleArea( pt, dst, org );
+static double DotProduct(Point2f u, Point2f v) {
+    return (double)u.x * v.x + (double)u.y * v.y;
+}
+
+static double Distance(Point2f a, Point2f b) {
+    return sqrt(DotProduct(b - a, b - a));
+}
+
+static bool Collinear(Point2f u, Point2f v) {
+    return abs(CrossProduct(u, v)) < FLT_EPSILON;
+}
+
+static bool InRectangle(Point2f x, Point2f a, Point2f b) {
+    return  min(a.x, b.x) < x.x && x.x < max(a.x, b.x) &&
+            min(a.y, b.y) < x.y && x.y < max(a.y, b.y);
+}
+
+static int CCW(Point2f a, Point2f b, Point2f c) {
```

Comment:
Could you set some readable name for the function?

Reply:
actually, this is fine. CCW is quite common notation for "counter-clockwise" in computational geometry. But probably there should be a short 1-line comment that explains wha the function does, e.g. "checks if the points a, b and c are a triangle vertices in counter-clockwise order"